statsmodels/distributions/joint_distribution.py

```diff
@@ -0,0 +1,526 @@
+
+        """
+
+    @abstractmethod
+    def pdf(self, x):
+        """Probability density function."""
+
+    def logpdf(self, x):
+        """Log of the PDF."""
+        return np.log(self.pdf(x))
+
+    @abstractmethod
+    def cdf(self, x):
+        """Cumulative density function."""
+
+    def plot(self, n, random_state=None, ax=None):
+        """Sample the copula and plot.
+
+        Parameters
+        ----------
+        n : int, optional
+            Number of samples to generate from the copula. Default is 1.
+        random_state : {None, int, `numpy.random.Generator`}, optional
+            If `seed` is None the `numpy.random.Generator` singleton is used.
+            If `seed` is an int, a new ``Generator`` instance is used,
+            seeded with `seed`.
+            If `seed` is already a ``Generator`` instance then that instance is
+            used.
+        ax : AxesSubplot, optional
+            If given, this subplot is used to plot in instead of a new figure
+            being created.
+
+        Returns
+        -------
+        fig : Figure
+            If `ax` is None, the created figure.  Otherwise the figure to which
+            `ax` is connected.
+        sample : array_like (n, d)
+            Sample from the copula.
+
+        """
+        sample = self.random(n=n, random_state=random_state)
+
+        fig, ax = utils.create_mpl_ax(ax)
+        ax.scatter(sample[:, 0], sample[:, 1])
+        ax.set_xlabel('u')
+        ax.set_ylabel('v')
+
+        return fig, sample
+
+    def plot_pdf(self, ticks_nbr=10, ax=None):
+        """Plot the PDF.
+
+        Parameters
+        ----------
+        ticks_nbr : int, optional
+            Number of color isolines for the PDF. Default is 10.
+        ax : AxesSubplot, optional
+            If given, this subplot is used to plot in instead of a new figure
+            being created.
+
+        Returns
+        -------
+        fig : Figure
+            If `ax` is None, the created figure.  Otherwise the figure to which
+            `ax` is connected.
+
+        """
+        n_samples = 100
+
+        xx, yy = np.meshgrid(np.linspace(0.0001, 1, n_samples),
+                             np.linspace(0.0001, 1, n_samples))
+        points = np.vstack([xx.ravel(), yy.ravel()]).T
+
+        data = self.pdf(points).T.reshape(xx.shape)
+        min_ = np.percentile(data, 5)
+        max_ = np.percentile(data, 95)
+
+        fig, ax = utils.create_mpl_ax(ax)
+
+        vticks = np.linspace(min_, max_, num=ticks_nbr)
+        range_cbar = [min_, max_]
+        cs = ax.contourf(xx, yy, data, vticks,
+                         antialiased=True, vmin=range_cbar[0],
+                         vmax=range_cbar[1])
+
+        ax.set_xlabel("u")
+        ax.set_ylabel("v")
+        ax.set_xlim(0, 1)
+        ax.set_ylim(0, 1)
+        ax.set_aspect('equal')
+        cbar = plt.colorbar(cs, ticks=vticks)
+        cbar.set_label('p')
+
+        return fig
+
+    @property
+    def tau(self):
```

Comment:
`tau` should be analytical or numerical integration

Reply:
I will have to do some research here. TBH my knowledge about all this is all here.